### ISSUE TITLE
Highlighted multilevel property functions as binary toggle

### ISSUE THREAD

**Issue by flatsiedatsie** [mozilla-iot/gateway#1906]
![centered-multilevel-toggles](https://user-images.githubusercontent.com/805405/58233836-a1513e80-7d2c-11e9-94f6-89ba3be56d69.gif)

When a multilevel property is highlighted, I expected to be able to set the value by clicking on a desired level inside the highlighted bubble.

Instead, clicking on the MultiLevelSwitch actually toggles the PowerPlug/OnOff property. 

This behaviour was unexpected. It seems to indicate some logic that if a thing also has a Powerplug/OnOff property, then that takes precedence?

it could also be an error on my part: perhaps things should not have multiple @type's?

**Comment by hobinjk** [comment]
In this case it's that the "Switch" part of "MultiLevelSwitch" is considered more important than "MultiLevel". This is essentially also why it goes from "18%" to "OFF" (it is a bug for it to be referring to the level property as a percentage here, I think you mentioned that in a different issue).

**Comment by flatsiedatsie** [comment]
I don't quite understand.

Wouldn't the switch functionality then result in the thermostat level going from 18 to 0 (and back?)

I don't understand how making a change in property A can have a result in property B. For me, conceptually, these are two totally unrelated properties.

Looking at the schema I guess you're saying that if I give a property the type 'MultiLevelSwitch', then this is actually something that goes for the entire thing. And the thing will look for two properties: one MultiLevel (compulsory) and one on/off switch (optional). If it finds an on/off switch, any on/off switch, it will pair it with the MultiLevel property.

I see how this 'marrrying' of two properties solved the issue of having a property that can be off, on, and a percentage at the same time. It's been outsourced to another switch!

@mrstegeman This could explain some of the [unexpected behaviour](https://github.com/mozilla-iot/gateway/issues/1907).



POSSIBLE SOLUTIONS

A. Perhaps in my case (building the MySensors add-on) one solution is to not ever add a type to a thing. I simply cannot fully predict which properties users will cobble together. But then you get the sad situation where the thing has no icon and no property that can be centered :-(

B. I could try to come up with some hierarchy and make sure the add-on only assigns one type to a thing. For example, if a thing has a toggle and a heating property, I could label it a thermostat type. But I worry about edge cases and angry Arduino programmers who want to know why they can't set their preferred property in the center. And it still leaves the problem that if a user add two toggles to the thing, it will lead to unexpected/interfering behaviour?

C. I could brashly suggest a change in a core concept of the Gateway :-) This is just me thinking out loud, but why try and model every possible device configuration in the Mozilla Gateway? It will inevitably lead to unexpected behaviour, edge cases and many demands to add all kinds of device types? Why not offload that task to users and add-on developers?

Add-on developers could connect things optimally for their situations. From this perspective properties are lego-pieces in a toolbox, a set of self-contained UI elements/properties which add-ons can pull together optimally. Perhaps an add-on can select which two properties should be married? Perhaps that's already possible.. 

Instead of making it dependent on the type, users could [choose which property they want centered](https://github.com/mozilla-iot/gateway/issues/1826) and which icon they would like the think to have.


I get why this has been done, but to me it seems this 'smart' behaviour is more work for you, and can actually be limiting to which property combinations can be made? Then again, perhaps overly-enthousiastic Arduino makers who piece together Frankenstein things were/are not the target audience ;-)

**Comment by benfrancis** [comment]
If I understood correctly, this is by design and is intended to make all the thing icons consistent in the things view. Tapping icons only turns devices on and off, to do anything else you need to go to the detail view.

I'd like to support the ability for developers to define custom capabilities and their UI via add-ons in future.

**Comment by hobinjk** [comment]
The switch functionality is to go from powered with setpoint 18 to unpowered with setpoint 18 so there is no actual interaction between the two properties, just a change in the display

**Comment by flatsiedatsie** [comment]
In the end I went with option B2. Now that I understand better how this system works I made the MySensors add-on much more conservative in how often it hands out a capability.

If it would become possible for users to choose any property as the 'central' one, then this issue will be moot, since the reason I tried to hand out capabilities as much as possible was so that properties could then easily take the center position (and have a cool icon).
